rollup.config.js

```diff
@@ -37,44 +39,89 @@ const resolveOpts = {
   extensions: [".ts", ".js"],
 };
 
-export default (async () => [
-  {
-    input: `./src/single-spa${isProduction ? "" : ".profile"}.ts`,
-    output: [
-      {
-        file: `./lib/umd/single-spa${isProduction ? ".min" : ".dev"}.cjs`,
-        format: "umd",
-        name: "singleSpa",
-        sourcemap: true,
-        banner: generateBanner("UMD"),
-      },
-      {
-        file: `./lib/system/single-spa${isProduction ? ".min" : ".dev"}.js`,
-        format: "system",
-        sourcemap: true,
-        banner: generateBanner("SystemJS"),
-      },
-      {
-        file: `./lib/esm/single-spa${isProduction ? ".min" : ".dev"}.js`,
-        format: "esm",
-        sourcemap: true,
-        banner: generateBanner("ESM"),
-      },
-    ],
-    plugins: [
-      replace(replaceOpts),
-      resolve(resolveOpts),
-      babel(babelOpts),
-      commonjs(),
-      isProduction &&
-        (await import("@rollup/plugin-terser")).default(terserOpts),
-      useAnalyzer && analyzer(),
-    ],
-  },
-])();
-
 function generateBanner(format) {
   return `/* single-spa@${packageJson.version} - ${format} - ${
     isProduction ? "prod" : "dev"
   } */`;
 }
+
+const commonPlugins = [
+  replace(replaceOpts),
+  resolve(resolveOpts),
+  babel(babelOpts),
+  commonjs(),
+  isProduction && (await import("@rollup/plugin-terser")).default(terserOpts),
+  useAnalyzer && analyzer(),
+];
+
+const commonConfiguration = {
+  input: `./src/single-spa${isProduction ? "" : ".profile"}.ts`,
+};
+
+export default async () => [
+  // UMD build
+  {
+    ...commonConfiguration,
+    output: {
+      file: `./lib/umd/single-spa${isProduction ? ".min" : ".dev"}.cjs`,
+      format: "umd",
+      name: "singleSpa",
+      sourcemap: true,
+      banner: generateBanner("UMD"),
+    },
+    plugins: [
+      ...commonPlugins,
+      typescript({
+        declarationDir: "./lib/umd/types-tsc",
```

Comment:
nitpick - `tsc` is the cli and we're not using the cli, so the name of this intermediate folder is a bit misleading

```suggestion
        declarationDir: "./lib/umd/types-unbundled",
```

Reply:
Agreed. It is used internally by the plugin, but this wording makes it more clear.